src/test/java/org/minions/devfund/danielmontecinos/moviedatabase/MovieDatabaseTest.java

```diff
@@ -0,0 +1,198 @@
+     */
+    @Test
+    public void testAddMovieMethodDoesNotAddSameActorsToActorList() {
+        movieDatabase.addMovie("Movie 9", new String[]{ACTOR_1, ACTOR_2});
+        movieDatabase.addMovie(MOVIE_7, new String[]{ACTOR_1, ACTOR_2});
+
+        final int expectedSize = 2;
+        assertEquals(expectedSize, movieDatabase.getActorList().size());
+    }
+
+    /**
+     * Tests addRating method adds a rating value to movie instance.
+     */
+    @Test
+    public void testAddRatingMethodAddsMovieRating() {
+        final double actualRating = 98.0;
+        movieDatabase.addMovie(MOVIE_7, new String[]{ACTOR_6});
+        movieDatabase.addRating(MOVIE_7, actualRating);
+
+        final double expectedRating = 98.0;
+        assertEquals(expectedRating, movieDatabase.getMovieList().get(0).getRating(), 0.0);
+    }
+
+    /**
+     * Tests that updateRating methods doesn't updated rating if value is same that actual rating.
+     */
+    @Test
+    public void testAddRatingMethodWhenValueIsSameThanTheUsedBefore() {
+        movieDatabase.addMovie(MOVIE_1, new String[] {ACTOR_1, ACTOR_2});
+
+        final double updatedRating = 0.0;
+        movieDatabase.updateRating(MOVIE_1, updatedRating);
+        assertEquals(updatedRating, movieDatabase.getMovieList().get(0).getRating(), 0.0);
+    }
+
+    /**
+     * Tests that updateRating methods updates rating previously set.
+     */
+    @Test
+    public void testUpdateRatingMethodUpdatesMovieRatingSetsBeforeUseIt() {
+        movieDatabase.addMovie(MOVIE_10, new String[]{ACTOR_10});
+        final double actualRating = 85.0;
+        movieDatabase.addRating(MOVIE_10, actualRating);
+
+        final double updatedRating = 100.0;
+        movieDatabase.updateRating(MOVIE_10, updatedRating);
+
+        final double expectedRating = 100.0;
+        assertEquals(expectedRating, movieDatabase.getMovieList().get(0).getRating(), 0.0);
+    }
+
+    /**
+     * Tests getBestActor method returns the actor with highest rating.
+     */
+    @Test
+    public void testGetBestActorReturnsTheActorWithHighestRating() {
+        final double midRating = 70.0;
+        movieDatabase.addMovie(MOVIE_1, new String[]{ACTOR_1, ACTOR_7});
+        movieDatabase.addRating(MOVIE_1, midRating);
+
+        final double worstRating = 20.0;
+        movieDatabase.addMovie(MOVIE_2, new String[]{ACTOR_9});
+        movieDatabase.addRating(MOVIE_2, worstRating);
+
+        final double bestRating = 100.0;
+        movieDatabase.addMovie(MOVIE_7, new String[]{ACTOR_5});
+        movieDatabase.addRating(MOVIE_7, bestRating);
+        assertEquals(ACTOR_5, movieDatabase.getBestActor());
+    }
+
+    /**
+     * Tests getBestMovie returns movie with highest rating.
+     */
+    @Test
+    public void testGetsBesMovieReturnsTheMovieWithHighestRating() {
+        final double bestRating = 100.0;
+        movieDatabase.addMovie("Best Movie", new String[]{ACTOR_1, ACTOR_2, ACTOR_21});
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Define a constant instead of duplicating this literal "Best Movie" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)